src/compiler/checker.ts

```diff
@@ -60,6 +72,35 @@ namespace ts {
         undefinedSymbol.declarations = [];
         const argumentsSymbol = createSymbol(SymbolFlags.Property | SymbolFlags.Transient, "arguments");
 
+        const {typeToString, getSymbolDisplayBuilder, symbolToString, signatureToString, typePredicateToString, visibilityToString} =
+            createWriter(
+                createSymbol,
+                getSymbolOfNode,
+                compilerOptions,
+                languageVersion,
+                getTypeParametersOfClassOrInterface,
+                getAccessibleSymbolChain,
+                needsQualification,
+                getQualifiedLeftMeaning,
+                hasExternalModuleSymbol,
+                isTypeAny,
+                () => nothingType,

```

Comment:
I'll wait on the interfaces until I've done a few more of these. I expect the interfaces will split up as they're used by multiple modules.